operations/kubernetes/csm-service-upgrade-FAILED.md

```diff
@@ -0,0 +1,193 @@
+## _timed out waiting for the condition on jobs/cray-dns-unbound-manager_
+
+
+This document show how to resolve issue when upgrade CSM 1.0.0-rc.2 to 1.0.11, running the csm-service-upgrade.sh script. .
+
+Items below will be part of this document:
+- Issue representation
+- Solution
+- Long Term Solution
+
+
+## Issue
+
+```sh
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Releasing cray-dns-unbound v0.4.12
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+2022-04-18T23:35:28Z INF Found value overrides for chart, applying:
+forwardZones:
+- forwardIps:
+  - 172.30.84.40
+  name: .
+global:
+  appVersion: 0.4.12
+ chart=cray-dns-unbound command=ship namespace=services version=0.4.12
+2022-04-18T23:35:28Z INF Running helm install/upgrade with arguments: upgrade --install cray-dns-unbound helm/cray-dns-unbound-0.4.12.tgz --namespace services --create-namespace --set global.chart.name=cray-dns-unbound --set global.chart.version=0.4.12 -f /tmp/loftsman-1650324903/cray-dns-unbound-values.yaml chart=cray-dns-unbound command=ship namespace=services version=0.4.12
+2022-04-18T23:36:36Z INF Release "cray-dns-unbound" has been upgraded. Happy Helming!
+NAME: cray-dns-unbound
+LAST DEPLOYED: Mon Apr 18 23:35:28 2022
+NAMESPACE: services
+STATUS: deployed
+REVISION: 23
+TEST SUITE: None
+ chart=cray-dns-unbound command=ship namespace=services version=0.4.12
+2022-04-18T23:36:36Z INF Ship status: success. Recording status, manifest to configmap loftsman-core-services in namespace loftsman command=ship
+2022-04-18T23:36:36Z INF Recording log data to configmap loftsman-core-services-ship-log in namespace loftsman command=ship
++ ./lib/wait-for-unbound.sh
++ kubectl wait -n services job cray-sls-init-load --for=condition=complete --timeout=20m
+job.batch/cray-sls-init-load condition met
++ kubectl wait -n services deployment cray-dns-unbound --for=condition=available --timeout=20m
+deployment.apps/cray-dns-unbound condition met
++ kubectl wait -n services job cray-dns-unbound-coredns --for=condition=complete --timeout=20m
+job.batch/cray-dns-unbound-coredns condition met
++ export -f poll-saw-completed-job
++ timeout 20m bash -c 'set -exo pipefail; poll-saw-completed-job'
++ poll-saw-completed-job
+++ jq '.items | length'
+++ kubectl get event -n services --field-selector involvedObject.kind=CronJob,involvedObject.name=cray-dns-unbound-manager,reason=SawCompletedJob -o json
++ [[ 2 -eq 0 ]]
++ kubectl wait -n services job -l cronjob-name=cray-dns-unbound-manager --for=condition=complete --timeout=5m
+job.batch/cray-dns-unbound-manager-1650322800 condition met
+job.batch/cray-dns-unbound-manager-1650322980 condition met
+job.batch/cray-dns-unbound-manager-1650323160 condition met
+timed out waiting for the condition on jobs/cray-dns-unbound-manager-1650323280
+timed out waiting for the condition on jobs/cray-dns-unbound-manager-1650324240
+[ERROR] - Unexpected errors, check output above
+/tmp
+CSM Service upgrade failed after 3 retries
+```
+
+
+## Solution
```

Comment:
General notes for this section:

- Command prompts should be `ncn-m001#`
- The content might flow better as a series of steps - see examples in comments below